packages/svelte/src/internal/client/runtime.js

```diff
@@ -239,6 +242,74 @@ export function check_dirtiness(reaction) {
 	return is_dirty;
 }
 
+/**
+ * @param {Error} error
+ * @param {import("#client").Effect} effect
+ * @param {import("#client").ComponentContext | null} component_context
+ */
+function handle_error(error, effect, component_context) {
+	// Given we don't yet have error boundaries, we will just always throw.
+	if (!DEV || handled_errors.has(error) || component_context === null) {
+		throw error;
+	}
+
+	const component_stack = [];
+
+	/** @type {import("#client").ComponentContext | null} */
+	let current_context = component_context;
+
+	while (current_context !== null) {
+		var filename = current_context.function?.filename;
+
+		if (filename) {
+			const file = filename.split('/').at(-1);
+			component_stack.push(file);
+		}
+
+		current_context = current_context.p;
+	}
+	const effect_name = effect.fn.name;
+
+	if (effect_name) {
+		// If we have an effect name, it means it happened in the template, $effect or $effect.pre.
+		Object.defineProperty(error, '__skip_hydration_retry', {
+			value: true,
+			enumerable: false
+		});
+	}
+
+	let component_stack_string =
+		`Svelte caught an error thrown by ${component_stack[0]}${effect_name ? ` during a${effect_name.startsWith('$') ? 'n' : ''} ${effect_name}` : ''}.` +
+		` You should fix this error in your code.\n\nError: ${error.message}\n\nThe component stack when this error occured:\n`;
+
+	for (const name of component_stack) {
+		component_stack_string += `\t${name}\n`;
+	}
+
+	component_stack_string += '\nThe error is located at:\n';
+
+	error.message = component_stack_string;
```

Comment:
I think we can cut back on the noise — it's weird to see 'Uncaught Error: Svelte caught an error ...' for example (was it uncaught or not?). Instead of this...

<img width="1020" alt="image" src="https://github.com/sveltejs/svelte/assets/1162160/df87a077-e25f-4cc9-9209-903452cdbbf9">

...what about just this?

<img width="1020" alt="image" src="https://github.com/sveltejs/svelte/assets/1162160/d59a75cf-05c3-4eb1-8f68-eb3fc218524a">


```suggestion
	const component_stack = [];

	const effect_name = effect.fn.name;

	if (effect_name) {
		// If we have an effect name, it means it happened in the template, $effect or $effect.pre.
		Object.defineProperty(error, '__skip_hydration_retry', {
			value: true,
			enumerable: false
		});

		component_stack.push(effect_name);
	}

	/** @type {import("#client").ComponentContext | null} */
	let current_context = component_context;

	while (current_context !== null) {
		var filename = current_context.function?.filename;

		if (filename) {
			const file = filename.split('/').at(-1);
			component_stack.push(file);
		}

		current_context = current_context.p;
	}

	const indent = /Firefox/.test(navigator.userAgent) ? '  ' : '\t';
	error.message += `\n${component_stack.map((name) => `\n${indent}in ${name}`).join('')}\n`;
```